### ISSUE TITLE
Onfido Passport Scanning Not Working on Cloud Device

### ISSUE THREAD

**Issue by harshitshah4** [onfido/onfido-ios-sdk#279]
We are trying to test Onfido Passport Scanning feature on Cloud iOS device, although the camera is injected with the passport image via sensor instrumentation, while scanning passport via Onfido, Its stuck on Camera screen.

Anyone who can help with detail about Onfido internals ? Will be helpful in debugging.

**Comment by andra-radu** [comment]
Hi @harshitshah4  - Apologies for the delay. Could you please how do you intend to test our SDK on a cloud device? what is your workflow, and what platform you are using?

**Comment by harshitshah4** [comment]
Hi @andra-radu, apologies from my end too, thanks for accommodating.
So, We are testing apps that are using Onfido SDK for Passport Scanning Feature Workflow on our BrowserStack platform, to test these apps, we are using the Image Injection feature ( https://www.browserstack.com/docs/app-automate/appium/advanced-features/camera-image-injection ).
TLDR; BrowserStack injects the image into the app, and the app receives that image as if the Camera is viewing that image. 

We observed that Onfido is using `AVCaptureVideoDataOutput` with BGRA pixel format and 1920x1080 preset for Camera Scanning in iOS, So we injected the image along with these configurations, But unfortunately, apps are not able to scan passports automatically, but if we click on the shutter button, the image is getting clicked correctly, which means that Image is getting injected. So we wanted to understand what Onfido does internally for automatically scanning the passport so that we are able to support apps that are using the Onfido Framework. 

Our best guesses were:
1. Liveness detection - High probable.
2. Detecting device orientation - Less probable.
3. Wants the passport to be in a very specific aspect ratio, or width and height - High probable or orientation.
4. AI ? - Not sure.

Some of the error messages we get ( Don't worry about the black screen, we just don't show the preview hence its black ) :
1. <img height="365" alt="Screenshot 2022-05-22 at 10 21 33 PM" src="https://user-images.githubusercontent.com/14928914/169706529-df7022bd-e9ba-4763-8e27-37357fab98c2.png">
2. <img height="365" alt="Screenshot 2022-05-22 at 10 41 42 PM" src="https://user-images.githubusercontent.com/14928914/169707236-59b458ce-2860-4280-9e23-af7ec352a1db.png">

But if we click on the shutter button, image seems to be injected correctly:
<img height="365" alt="Screenshot 2022-05-22 at 10 34 09 PM" src="https://user-images.githubusercontent.com/14928914/169706965-7e8f3b70-e5ea-4241-bd97-d9644f4992ae.png">

But, not sure why its not working automatically, without clicking the shutter button.

I know this seems like an issue from our end, but it would be really helpful if you could hint us on things Onfido does for Passport scanning so that we are able to support apps using Onfido to test on our platform or atleast if you can share what all things Onfido expects for Passport Scanning.

Please let me know if this is not clear or need more data.

**Comment by vguerci** [comment]
Hello, (auto)capture is automatically triggered when specific conditions are met, like image quality validations that are run on device such as detecting the document or that there is no glare on it, which are the kind of errors that you are getting. These validations are run continuously on the video stream, which I guess is not handling well your image injection.
Manually triggering the shutter button workaround this auto capture logic.
Does this answer your question?

**Comment by harshitshah4** [comment]
Can you help me list of such checks that are ran ?

**Comment by vguerci** [comment]
Sorry, I'm afraid I can't disclose the internals of what we are doing. Also, even if I could, these could change at any time, or based on many factors.

Would adding a public API for testing purposes to simulate / trigger a capture would work for you?
What are you trying to achieve precisely?

**Comment by vguerci** [comment]
Closing for now, feel free to reopen if needed, thank you.
